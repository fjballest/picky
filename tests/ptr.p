/*
!../8.pc  -P ptr.p && ../8.pi  -XDMS p.out >[2=1]
 */
program ptr;

types:
	Pint =  ^int;

procedure main()
	p: Pint;
	q: Pint;
{
	new(p);
	new(q);
	stack();
	data();
	p^ = 3;
	q = p;
	writeln(q^);
	stack();
	data();
}
