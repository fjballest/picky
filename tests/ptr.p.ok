stack trace at:
main() pid 0	pc 0x00000e ptr.p:15
	local variables:
	p = 0x401aba0
	q = 0x401ac00

global variables:
Maxint = 2147483647
Minint = 0
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
0x401ac00 = 1852401771
0x401aba0 = -825636661
3
stack trace at:
main() pid 0	pc 0x000030 ptr.p:20
	local variables:
	p = 0x401aba0
	q = 0x401aba0

global variables:
Maxint = 2147483647
Minint = 0
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
0x401ac00 = 1852401771
0x401aba0 = 3
memory leaks:
ptr.p:13	(1 times)
ptr.p:14	(1 times)
