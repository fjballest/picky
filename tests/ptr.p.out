stack trace at:
main() pid 0	pc 0x00000e ptr.p:15
	local variables:
	p = 0x401ac38
	q = 0x401ac98

global variables:
Maxint = 2147483647
Minint = -2147483646
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
0x401ac98 = -151916301
0x401ac38 = 1448169555
3
stack trace at:
main() pid 0	pc 0x000030 ptr.p:20
	local variables:
	p = 0x401ac38
	q = 0x401ac38

global variables:
Maxint = 2147483647
Minint = -2147483646
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
0x401ac98 = -151916301
0x401ac38 = 3
memory leaks:
ptr.p:13	(1 times)
ptr.p:14	(1 times)
